### PR TITLE
Feature: Scheduled timers

### PR DESCRIPTION
**Pull request by capsj**
### Description

This PR introduces a feature enabling actors to schedule the sending of periodic or delayed messages to itself by mixing in the `Timers` trait. The API and functionality follow Akka’s implementation (see [Akka’s docs](https://doc.akka.io/docs/akka/current/actors.html#actors-timers)).

### CODE REVIEW

File: src/main/scala/com/suprnation/actor/Timers.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    timers.cancelAll >> preRestart(reason, message)
+
+  override def aroundPostStop(): F[Unit] =
+    timers.cancelAll >> postStop
+
+  override def aroundReceive(receive: Receive[F, Request], msg: Any): F[Any] =
+    msg match {
+      case t: Timer[F, Request, Key] => _timers.interceptTimerMsg(t).ifM(receive(t.msg), unhandled(t.msg))
+      case _ => receive.applyOrElse(msg.asInstanceOf[Request], unhandled)
+    }
```
Comment (capsj) [inline src/main/scala/com/suprnation/actor/Timers.scala]: Not sure about these. I wanted to follow Akka and do:
```
 override def aroundPreRestart(reason: Option[Throwable], message: Option[Any]): F[Unit] =
    timers.cancelAll >> super.preRestart(reason, message)
```

but having `Timers extends Actor` feels wrong as well

Comment (PetrosPapapa) [reply src/main/scala/com/suprnation/actor/Timers.scala]: I don't think you have any option here as overriding and calling the self method won't work.

**Review by PetrosPapapa — APPROVED** [review]
Looks great to me! (pending the change to extend the Actor)

**Comment by capsj** [comment]
Merging into https://github.com/PetrosPapapa/cats-actors/pull/11 to create a release branch

**Pull request closed without merging**